app/main.py

```diff
@@ -2,0 +1,29 @@
+class Animal:
+    alive = []
+
+    def __init__(self, name: str) -> None:
+        self.name = name
+        self.health = 100
+        self.hidden = False
+        Animal.alive.append(self)
+
+    def __del__(self) -> None:
+        Animal.alive.remove(self)
+
+    def __repr__(self) -> None:
+        return (f"{{Name: {self.name}, "
+                f" Health: {self.health}, "
+                f" Hidden: {self.hidden}}}")
+
+
+class Herbivore(Animal):
+    def hide(self) -> None:
+        self.hidden = not self.hidden
+
+
+class Carnivore(Animal):
+    def bite(self, herbivore: Animal) -> None:
+        if isinstance(herbivore, Herbivore) and not herbivore.hidden:
+            herbivore.health -= 50
+            if herbivore.health <= 0:
+                del herbivore
```

Comment:
When a `Herbivore`'s health drops to zero or below, it should be removed from the `Animal.alive` list. Ensure that this is handled correctly, as simply deleting the object may not automatically remove it from the list.